content/post/108-js-ipfs-0.50.0.md

```diff
@@ -0,0 +1,201 @@
+---
+date: 2020-09-14
+url: /2020-09-14-js-ipfs-0-50/
+title: js-IPFS 0.50.0 runs in shared webworkers and has a faster ipfs.add
+header_image: js-ipfs-placeholder.png
+author: Alex Potsides
+---
+
+# 🔦 Highlights
+
+> share an IPFS node between multiple tabs and add files more quickly
+
+`js-IPFS@0.50.0` has landed with the ability to share a node between multiple browser tabs and greatly improved pinning performance.
+
+We're also phasing out use of Node.js Buffers as data types in favour of standard JavaScript Uint8Arrays.
+
+Read on for the full details!
+
+## 🤝 Share a node between browser tabs
+
+An IPFS node makes lots of connections to other nodes on the network, and more so since [delegate nodes were turned on by default](https://blog.ipfs.io/2020-07-20-js-ipfs-0-48/). This is to ensure you have the greatest chance of finding content on the network, and so other people have the greatest chance of finding your content on the network.
+
+This does not come without a price though, maintaining multiple connections can be resource-intensive and in some cases the browser will limit the number of concurrent connections you can have have.
+
+This can be a problem in web browsers if the user opens your app in two tabs, suddenly you have two nodes running with twice the number of open connections.  Worse, they are sharing a datastore and the same peer ID.
+
+Help is at hand in the shape of the [ipfs-message-port-client](https://www.npmjs.com/package/ipfs-message-port-client) and [ipfs-message-port-server](https://www.npmjs.com/package/ipfs-message-port-server) which allow you to run one IPFS node in a [SharedWorker](https://developer.mozilla.org/en-US/docs/Web/API/SharedWorker) and share it between multiple tabs within your application.
+
+There will be a more in depth post here on this subject soon but in the meantime check out the [browser-sharing-node-across-tabs](https://github.com/ipfs/js-ipfs/tree/master/examples/browser-sharing-node-across-tabs) for an example of how to use it!
+
+## 📌 Pins in the datastore
```

Comment:
The title of the post mentions ipfs.add() but it's not explicitly tied into this paragraph. Might be worth adding a sentence or somehow just making super explicit that add() does an implicit pin (if that's the case).

Reply:
I've changed the title to be more about pinning.